src/Spice86.Aeon/Emulator/Video/Dac.cs

```diff
@@ -70,70 +64,66 @@ public byte WriteIndex
         /// <returns>Red, green, or blue channel value.</returns>
         public byte Read()
         {
-            unsafe
+            Rgb color = _palette[_readIndex];
+            _readChannel++;
+            switch (_readChannel)
             {
-                uint color = palette[readIndex];
-                readChannel++;
-                if (readChannel == 1)
-                {
-                    return (byte)((color >> 18) & 0x3F);
-                }
-
-                if (readChannel == 2)
-                {
-                    return (byte)((color >> 10) & 0x3F);
-                }
-
-                readChannel = 0;
-                readIndex++;
-                return (byte)((color >> 2) & 0x3F);
+                case 1:
+                    return color.G;
+                case 2:
+                    return color.B;
             }
+
+            _readChannel = 0;
+            _readIndex++;
+            return color.R;
```

Comment:
```suggestion
                    return color.R;
                case 2:
                    return color.G;
            }

            _readChannel = 0;
            _readIndex++;
            return color.B;
```
_readChannel is updated before being used, so 1 is the first color (R)